labgrid/driver/rawnetworkinterfacedriver.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=no-member
+import json
+import signal
+import subprocess
+import shutil
+import tempfile
+
+import attr
+
+from .common import Driver
+from ..factory import target_factory
+from ..step import step
+from ..util.helper import processwrapper
+from ..util.managedfile import ManagedFile
+from ..resource.common import NetworkResource
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class RawNetworkInterfaceDriver(Driver):
+    bindings = {
+        "iface": {"NetworkInterface", "RemoteNetworkInterface", "USBNetworkInterface"},
+    }
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.proc_pcap_map = {}
+
+    def _get_wrapper_prefix(self):
+        return self.iface.command_prefix + ["sudo", "labgrid-raw-interface"]
+
+    @Driver.check_active
+    @step(args=["count"])
+    def record(self, *, count=None):
+        capture = tempfile.NamedTemporaryFile(prefix="lg-raw-int-cap-")
+        cmd = self._get_wrapper_prefix() + ["tcpdump", self.iface.ifname]
+        if count is not None:
+            cmd.append(str(count))
+
+        proc = subprocess.Popen(cmd, stdout=capture, stderr=subprocess.PIPE)
+        self.proc_pcap_map[proc] = capture
+        return proc
+
+    @Driver.check_active
+    @step(args=["proc", "filename"])
+    def get_record(self, proc, filename, *, timeout=None):
+        assert proc in self.proc_pcap_map
+
+        try:
+            _, err = proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            proc.send_signal(signal.SIGINT)
+            _, err = proc.communicate()
+
+        assert proc.returncode == 0, f"tcpdump err={err}"
```

Comment:
Reuse this as an internal helper for `stop_record`/`stop_replay`. Those would be convenience methods for easier use in test cases.